### ISSUE TITLE
'ReferenceError: Cannot access uninitialized variable.' in constructor

### ISSUE THREAD

**Issue by theakman2** [oven-sh/bun#1876]
### What version of Bun is running?

0.5.1

### What platform is your computer?

Linux 5.19.0-29-generic x86_64 x86_64

### What steps can reproduce the bug?

Create the following typescript file:

```ts
class Base {
	public value: number;
	constructor(value: number) {
		this.value = value;
	}
};

class Sub extends Base {
	constructor(value: number) {
		const newValue = value * 10;
		super(newValue);
	}
};

const sub = new Sub(123);

console.log(sub.value);
```

2. Execute the file with bun.

### What is the expected behavior?

The script should execute successfully and output '1230'.

### What do you see instead?

```
a@a:/tmp$ bun test.ts 
 6 | };
 7 | 
 8 | class Sub extends Base {
 9 |    constructor(value: number) {
10 |            const newValue = value * 10;
11 |            super(newValue);
     ^
ReferenceError: Cannot access uninitialized variable.
      at new Sub (/tmp/test.ts:11:2)
      at /tmp/test.ts:15:12
```

### Additional information

The file successfully compiles and executes with `tsc`.

**Comment by theakman2** [comment]
If anyone else is running into this problem, this seems to be a workaround:

```ts
class Sub extends Base {
	constructor(value: number) {
		super((() => {
			const newValue = value * 10;
			return newValue;
		})());
	}
};
```

**Comment by dylan-conway** [comment]
fixed with #1883
